tests/bluechi_test/machine.py

```diff
@@ -312,18 +312,33 @@ def __init__(
         self.copy_machine_lib()
 
 
-class BluechiControllerMachine(BluechiMachine):
-    def __init__(
-        self, name: str, client: Client, ctrl_config: BluechiControllerConfig
-    ) -> None:
-        super().__init__(name, client)
+class BluechiControllerMachine(BluechiAgentMachine):
+
+    agent_uds_address = "unix:path=/run/bluechi/bluechi.sock"
+    agent_uds_name = "node-local"
```

Comment:
We want to use UDS as default, so I think this should be moved to the bluechi agent config fixture:
https://github.com/eclipse-bluechi/bluechi/blob/main/tests/bluechi_test/fixtures.py#L206-L212

Reply:
Isn't this method used for standalone agents, which should continue to work as is? The local socket connection will be valid only for agent running on controller node, right?

Reply:
If we define the UDS as the default approach, then this should be reflected in the fixture as well. We can use two - one for uds and one for tcp. I wouldn't hard-code the local agent into the machine class.

Reply:
I mean, we can implement this in various ways. But we most likely have to refactor current test code since we have to ensure to start only as many agents (incl. the local one) as necessary - otherwise its pointless. 

At the same time we should evaluate if some tests still make sense with the local agent as default - for example, we have [test_bluechi_and_agent_on_same_machine](https://github.com/eclipse-bluechi/bluechi/blob/main/tests/tests/tier0/bluechi-and-agent-on-same-machine/test_bluechi_and_agent_on_same_machine.py), which doesn't make much sense to keep in my point of view.

Reply:
> At the moment, we spin up two agents: one on the controller and the other on a dedicated node (so the number of nodes hasn't been reduced).

In my understanding, the purpose of these changes is mainly to reduce the node count in multi-host testing(currently use 4 nodes). The expected result is to reduce from using 4 hosts to 2.

Reply:
> In my understanding, the purpose of these changes is mainly to reduce the node count in multi-host testing(currently use 4 nodes). The expected result is to reduce from using 4 hosts to 2.

Exactly. And at the moment this PR adds one agent, so we have n+1, and keeps the same number of nodes.

Reply:
IIUC the purpose of this PR is just start a local agent on controller node without touching the rest. The reducing part is for subsequent PRs